src/FileStateRegistry.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\FtpExtractor;
+
+use Symfony\Component\Filesystem\Filesystem;
+
+class FileStateRegistry
```

Comment:
Zabalení té logiky je fajn, ale myslím, že čtení/zápis statefile by měla být zodpovědnost komponenty (od toho konec koncu už má metodu [`getInputState`](https://github.com/keboola/php-component/blob/master/src/BaseComponent.php#L158) a [`loadInputState`](https://github.com/keboola/php-component/blob/master/src/BaseComponent.php#L91). Takže by se mi víc líbilo, kdyby se ten výsledný state vracel a zapisoval až v komponentě (ideálně můžeš udělat PR do php-component na metodu na zpětný zápis stateFile).

Reply:
Mimo jiné to porušuje SRP - řeší to logiku okolo souborů a současně to zapisuje/čte výsledek.

Reply:
Nevěděl jsem, že funguje getInputState. Existuje i writeOutputStateToFile a tu jsem použil pro zápis.